src/js/components/VoterGuide/GuideList.jsx

```diff
@@ -49,18 +49,26 @@ export default class GuideList extends PureComponent {
     const voterGuideList = this.sortOrganizations(nextProps.incomingVoterGuideList, ballotItemWeVoteId);
     this.setState({
       voterGuideList,
-      ballotItemWeVoteId: nextProps.ballotItemWeVoteId,
+      ballotItemWeVoteId: this.props.ballotItemWeVoteId,
     }, () => {
       const orgsWithPositions = this.getOrganizationsWithPositions();
-      // this.setState({
-      //   organizationsWithPositions: orgsWithPositions,
-      // });
-      if (!this.state.filteredOrganizationsWithPositions || !this.state.filteredOrganizationsWithPositions.length) {
-        this.setState({ filteredOrganizationsWithPositions: orgsWithPositions });
-      }
+      this.setState({
+        // organizationsWithPositions: orgsWithPositions,
+        filteredOrganizationsWithPositions: orgsWithPositions,
+      });
     });
   }
 
+  shouldComponentUpdate (nextState) {
+    if (this.state.voterGuideList !== nextState.voterGuideList) {
```

Comment:
With lists and objects, React typically does a shallow compare, which I think would not find voterGuideList had changed even if list items had changed. I found this article that suggests using a function that compares the length of the lists, and other quick approaches: https://developmentarc.gitbooks.io/react-indepth/content/life_cycle/update/using_should_component_update.html